src/gui/UBMagnifer.cpp

```diff
@@ -194,11 +194,11 @@ void UBMagnifier::calculateButtonsPositions()
     sDecreasePixmapButtonRect = QRect(sChangeModePixmapButtonRect.x() - sChangeModePixmap->width() - m_iButtonInterval, size().height() - sDecreasePixmap->height(), sDecreasePixmap->width(), sDecreasePixmap->height());
     sIncreasePixmapButtonRect = QRect(sDecreasePixmapButtonRect.x() - sChangeModePixmap->width() - m_iButtonInterval, size().height() - sIncreasePixmap->height(), sIncreasePixmap->width(), sIncreasePixmap->height());
 
-    qDebug() << "mResizeItemButtonRect" << mResizeItemButtonRect << endl
-             << "sClosePixmapButtonRect" << sClosePixmapButtonRect << endl
-             << "sChangeModePixmapButtonRect" << sChangeModePixmapButtonRect << endl
-             << "sDecreasePixmapButtonRect" << sDecreasePixmapButtonRect << endl
-             << "sIncreasePixmapButtonRect" << sIncreasePixmapButtonRect << endl;
+    qDebug() << "mResizeItemButtonRect" << mResizeItemButtonRect << '\n'
```

Comment:
Replacing endl with '\n' is not necessary, it has simply been moved into the Qt namespace to not conflict with std::endl.
So Qt::endl would work.

Reply:
Seems that I do not use the right search terms, but I cannot find the definition of `Qt::endl` either in the Qt documentation nor by grepping the include directories. Can you help me?

And I remember that again for my old Qt 5.12 this definition of `Qt::endl` was not available, so I chose the "compatible" way.

Reply:
Here is the documentation for it: https://doc.qt.io/qt-5/qt.html#endl
Seems like it was introduced in Qt 5.14.

Reply:
Yeah, I had to read some pages explaining how stream modifiers actually work and how just specifying their name invokes them on the stream, but I hope I now have understood that. And yes, your porposal looks fine. Will look on it tomorrow. Note that I'm on the openSUSE conference in Nuremberg from Wednesday to Saturday, so I might not be responsive during this time.

Reply:
My approach would now be to just define the missing `Qt::endl` for Qt < 5.14:
```c++
#if QT_VERSION < QT_VERSION_CHECK(5, 14, 0)
namespace Qt 
{
inline QTextStream &endl(QTextStream &s) {
    return ::endl(s);
}
#endif
```
When dropping Qt < 5.14 compatibility just this definition will be dropped, the code needs no adaptation.

Reply:
According to [flush's documentation](https://doc.qt.io/qt-5/qtextstream.html#flush), if QTextStream operates on a string, flush does nothing, so I don't think there is a slight performance win on these cases. But preprocessor  directives seem avoidable with endl doing `return stream << '\n'_L1 << Qt::flush;` ([source](https://code.qt.io/cgit/qt/qtbase.git/tree/src/corelib/serialization/qtextstream.cpp?h=dev)). 

So I would personally prefer @letsfindaway's solution. I just wonder what's the reason to use a QChar1Latin. Is it related to `Note: On Windows, all '\\n' characters are written as '\\r\\n' if
    QTextStream's device or string is opened using the QIODevice::Text flag.` described in endl's documentation ?

Reply:
We're operating on I/O streams (with `qDebug()`) or files (in `UBGraphicsWidgetItem.cpp:1146ff`), so flushing affects performance.

And for the `_L1` appendix: this is rather new, I thing with upcoming Qt 6.4, so we don't have to care about it currently. I think it just makes clear that this character is 8 bits and shall remain 8 bits.

Reply:
oops sry I read too fast the line you had pointed and just saw `"<html>..." << endl`

Reply:
> We're operating on I/O streams (with `qDebug()`) ... so flushing affects performance.

I thought qDebug() calls were totally ignored in release mode, but now that I think of it I don't know how exactly. Is there a way they would be ignored or go to a "do nothing" directive but the flush still performed on an empty stream ? Would this affect performance in a release context ? If not, it should not bother us that much...

> And for the `_L1` appendix: this is rather new, I thing with upcoming Qt 6.4, so we don't have to care about it currently. I think it just makes clear that this character is 8 bits and shall remain 8 bits.

The `_L1` syntax could be rather new, but there was a QLatin1Char instanciation instead in the previous Qt versions (I suppose the "character litteral" syntax  just saves some typing ?), so I still wonder why it is done. Could it be important to maintain the use of `endl` on files to handle OS-specific end of lines ?

Reply:
I still assume that it does not matter. What happens:

- using `<< '\n'` the `QTextStream &operator<<(char ch)` will be used. Internally this is implemented as:
```c++
QTextStream &QTextStream::operator<<(char c)
{
    Q_D(QTextStream);
    CHECK_VALID_STREAM(*this);
    d->putChar(QChar::fromLatin1(c));
    return *this;
}
```
So it interprets the argument as Latin1, converts it to a `QChar` and then puts it to the stream.

- using `QLatin1Char('\n')` there is no explicit stream operator for that type, so the implicit conversion operator of `QChar` will be used to convert it and then use `QTextStream &operator<<(QChar ch)`. You can imagine how this implementation looks like:
```c++
QTextStream &QTextStream::operator<<(QChar c)
{
    Q_D(QTextStream);
    CHECK_VALID_STREAM(*this);
    d->putChar(c);
    return *this;
}
```

In both cases the same conversion steps are performed. The handling of the platform specific line endings is handled in the underlying `QIODevice`, not in `QTextStream`.